### ISSUE TITLE
Reduce false positives for remark/no-dead-urls

### ISSUE THREAD

**Issue by transitive-bullshit** [sindresorhus/awesome-lint#42]
Continuing the discussion from #38.

Repro case with multiple false positives: [https://travis-ci.org/carlosperate/awesome-microbit/builds/423808998](https://travis-ci.org/carlosperate/awesome-microbit/builds/423808998)

cc @carlosperate @davidtheclark

**Comment by transitive-bullshit** [comment]
Note that these all repro for me locally, so my guess is it's a problem with the way `check-links` is invoking `got`.

**Comment by sindresorhus** [comment]
> ✖ 182:3 Link to http://wiki.freepascal.org/micro:bit is dead no-dead-urls

Seems like many of these are actually server errors. For example:

```
❯ curl http://wiki.freepascal.org/micro:bit
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<!--< html xmlns="http://www.w3.org/1999/xhtml">-->
<head>
<title>HTTP Error 400</title>
</head>
<body>
<h1>Error 400</h1>
<p>We're sorry, but we could not fulfill your request for
/micro:bit on this server.</p>
<p>An invalid request was received from your browser. This may be caused by a malfunctioning proxy server or browser privacy software.</p>
<p>Your technical support key is: <strong>3d5a-06fd-6992-0ee5</strong></p>
<p>You can use this key to <a href="http://www.ioerror.us/bb2-support-key?key=3d5a-06fd-6992-0ee5">fix this problem yourself</a>.</p>
<p>If you are unable to fix the problem yourself, please contact <a href="mailto:vincent.snijders+nospam@nospam.gmail.com">vincent.snijders at gmail.com</a> and be sure to provide the technical support key shown above.</p>
```

**Comment by sindresorhus** [comment]
However, `wget` is able to download the page. It seems cookies are required to make it work. Notice the `Set-Cookie: bb2_screener_=1536004423+61.90.6.253; path=//` here:

```
❯ wget http://wiki.freepascal.org/micro:bit
--2018-09-04 02:53:42--  http://wiki.freepascal.org/micro:bit
Resolving wiki.freepascal.org (wiki.freepascal.org)... 193.224.143.39
Connecting to wiki.freepascal.org (wiki.freepascal.org)|193.224.143.39|:80... connected.
HTTP request sent, awaiting response...
  HTTP/1.1 200 OK
  Date: Mon, 03 Sep 2018 19:53:43 GMT
  Server: Apache/2.4.18 (Ubuntu) OpenSSL/1.0.2g
  X-Content-Type-Options: nosniff
  Set-Cookie: bb2_screener_=1536004423+61.90.6.253; path=//
  Vary: Accept-Encoding,Cookie
  Expires: Thu, 01 Jan 1970 00:00:00 GMT
  Cache-Control: private, must-revalidate, max-age=0
  Last-Modified: Wed, 19 Apr 2017 02:22:15 GMT
  Content-Language: en
  Keep-Alive: timeout=5, max=100
  Connection: Keep-Alive
  Transfer-Encoding: chunked
  Content-Type: text/html; charset=UTF-8
Length: unspecified [text/html]
Saving to: ‘micro:bit.html’

micro:bit.html                              [  <=>                                                                         ]  13.94K  54.4KB/s    in 0.3s

2018-09-04 02:53:43 (54.4 KB/s) - ‘micro:bit.html’ saved [14279]
```

**Comment by sindresorhus** [comment]
Got just got (pun intended) support for cookies: https://github.com/sindresorhus/got#cookies

**Comment by transitive-bullshit** [comment]
I've updated our downstream dependencies [check-links](https://github.com/transitive-bullshit/check-links) and [remark-lint-no-dead-urls](https://github.com/davidtheclark/remark-lint-no-dead-urls/pull/9) to make the URL resolution more robust.

@sindresorhus this ended up being unrelated to cookies. I had to [enable](https://github.com/transitive-bullshit/check-links/blob/master/lib/check-link.js#L27-L39) more default headers for some of these links and `rejectUnauthorized` solved the issues for two of them. Oddly enough, several of these links required a valid `accept-language` header to be sent otherwise they resulted in 500 errors.

In general, I think we should err on the side of being as lenient as possible for these types of cases to reduce false positives.

@carlosperate the only remaining link that fails (after reinstalling all deps while we wait for the PRs to propagate), is `http://make.techwillsaveus.com/bbc-microbit`. This URL leads to a 302 => 404 with an HTML page that using JavaScript redirects to `https://make.techwillsaveus.com/bbc-microbit`. Given that it's using JavaScript to perform the redirect, I think it's safe to say that you should just update this link instead of us trying to support this use case.
